frontend/src/components/Pagination.jsx

```diff
@@ -30,8 +30,10 @@ export const Pagination = ({
   const { page: paramPageOneBased = 1 } = useParams();
 
   useEffect(() => {
-    setActivePage(Page.fromOneBased(parseInt(paramPageOneBased)));
-  }, [paramPageOneBased]);
+    const newActivePage = Page.fromOneBased(parseInt(paramPageOneBased));
+    setActivePage(newActivePage);
+    fetchPage(newActivePage);
+  }, [paramPageOneBased, fetchPage]);
```

Comment:
I need to look at it a bit more.

While it makes BACK work every time, it causes the initial request after page loading to be sent twice: 1. from parent component, 2. from `Pagination`.